src/main/java/de/metas/ui/web/pickingV2/packageable/process/PackageablesView_PrintPicklist.java

```diff
@@ -95,18 +101,32 @@ protected String doIt()
 	{
 		final PackageableRow row = getSingleSelectedRow();
 
-		createPickingCandidatesIfNeeded(row);
 
-		// print
-		final byte[] pickList = printPicklist(row);
 
-		// preview
-		getResult().setReportData(
-				pickList,
-				buildFilename(row),
-				OutputType.PDF.getContentType());
+		final ShipmentScheduleLockRequest lockRequest = createLockRequest(row);
+		locksRepo.lock(lockRequest);
+
+		try
+		{
+			createPickingCandidatesIfNeeded(row);
+
+			// print
+			final byte[] pickList = printPicklist(row);
 
-		return MSG_OK;
+			// preview
+			getResult().setReportData(
+					pickList,
+					buildFilename(row),
+					OutputType.PDF.getContentType());
+
+			return MSG_OK;
+		}
+		catch (final Exception ex)
+		{
+			locksRepo.unlockNoFail(ShipmentScheduleUnLockRequest.of(lockRequest));
```

Comment:
* i think this needs to be in final, or how/where is it unlocked in case the exception is not thrown
* if it needs to be unlocked in final (i.e. here), then i suggest to use try-with-ressouces

Reply:
the user will unlock it manually